zipline/finance/commission.py

```diff
@@ -12,113 +12,162 @@
 
     Parameters
     ----------
     cost : float, optional
-        The flat amount of commisions paid per trade.
+        The flat amount of commissions paid per trade.
     """
 
     def __init__(self, cost=DEFAULT_MINIMUM_COST_PER_TRADE):
         """
-        Cost parameter is the cost of a trade, regardless of
-        share count. $5.00 per trade is fairly typical of
-        discount brokers.
+        Cost parameter is the cost of a trade, regardless of share count.
+        $5.00 per trade is fairly typical of discount brokers.
         """
         # Cost needs to be floating point so that calculation using division
         # logic does not floor to an integer.
         self.cost = float(cost)
 
-    def calculate(self, transaction):
+    def calculate(self, order, transaction):
         """
-        returns a tuple of:
-        (per share commission, total transaction commission)
+        If the order hasn't had a commission paid yet, pay the fixed
+        commission.
         """
-        if transaction.amount == 0:
-            return 0.0, 0.0
-
-        return abs(self.cost / transaction.amount), self.cost
+        if order.commission == 0:
+            # if the order hasn't had a commission attributed to it yet,
+            # that's what we need to pay.
+            return self.cost
+        else:
+            # order has already had commission attributed, so no more
+            # commission.
+            return 0.0
 
 
-class PerDollar(object):
-    """Calculates a commission for a transaction based on a per
-    dollar cost.
+class PerDollar(CommissionModelBase):
+    """
+    Calculates a commission for a transaction based on a per trade cost.
 
     Parameters
     ----------
-    cost : float, optional
-        The amount of commissions paid per dollar traded.
+    cost : float
+        The flat amount of commissions paid per trade.
     """
 
     def __init__(self, cost=0.0015):
         """
         Cost parameter is the cost of a trade per-dollar. 0.0015
-        on $1 million means $1,500 commission (=1,000,000 x 0.0015)
+        on $1 million means $1,500 commission (=1M * 0.0015)
         """
-        self.cost = float(cost)
+        self.cost_per_dollar = float(cost)
 
     def __repr__(self):
-        return "{class_name}(cost={cost})".format(
+        return "{class_name}(cost_per_dollar={cost})".format(
             class_name=self.__class__.__name__,
-            cost=self.cost)
+            cost=self.cost_per_dollar)
 
-    def calculate(self, transaction):
+    def calculate(self, order, transaction):
         """
-        returns a tuple of:
-        (per share commission, total transaction commission)
+        Pay commission based on dollar value of shares.
         """
-        cost_per_share = transaction.price * self.cost
-        return cost_per_share, abs(transaction.amount) * cost_per_share
+        cost_per_share = transaction.price * self.cost_per_dollar
```

Comment:
Curious: does the commission get calculated `before or after slippage has been applied?